core/src/impl/Kokkos_Stacktrace.cpp

```diff
@@ -168,61 +167,46 @@ main_column_info find_main_column(const std::vector<std::string>& traceback) {
     }
   }
 
-  // Make another pass to get the column lengths.
-  // Only demangle the column of functions.
-  std::vector<size_t> max_col_lengths;
-  for (auto&& entry : traceback) {
-    size_t col_count = 0;
-    for_each_token(entry, [&](const std::string& s, bool) {
-      const size_t cur_col_len =
-          (found_main && col_count == main_col) ? demangle(s).size() : s.size();
-      ++col_count;
-      if (max_col_lengths.size() < col_count) {
-        max_col_lengths.push_back(cur_col_len);
-      } else {
-        const size_t old_max_len = max_col_lengths[col_count - 1];
-        if (old_max_len < cur_col_len) {
-          max_col_lengths[col_count - 1] = cur_col_len;
-        }
-      }
-    });
-  }
-  return main_column_info{found_main, main_col, max_col_lengths};
+  return main_column_info{found_main, main_col};
 }
 
-void demangle_and_print_traceback_entry(
-    std::ostream& out, const std::string& traceback_entry,
-    const bool found_main, const size_t main_col,
-    const std::vector<size_t>& max_col_lens) {
+void demangle_and_print_traceback_entry(std::ostream& out,
+                                        const std::string& traceback_entry,
+                                        const bool found_main,
+                                        const size_t main_col) {
   std::vector<std::string> tokens;
   size_t cur_col = 0;
+
+  // Print the address column first
   for_each_token(traceback_entry, [&](const std::string& s, bool last) {
-    const size_t old_width(out.width());
-    out.width(max_col_lens[cur_col]);
-    try {
-      if (found_main && cur_col == main_col) {
-        out << demangle(s);
-      } else {
-        out << s;
-      }
-      if (!last) {
-        out << " ";
-      }
-      ++cur_col;
-    } catch (...) {
-      out.width(old_width);
-      throw;
+    if (!(found_main && cur_col == main_col)) {
+      out << s;
+    }
+    if (!last) {
+      out << " ";
```

Comment:
We can probably simplify even further and not worry about an extra space in the last line:
```suggestion
      out << s << " ";
```

This would get rid of `bool last` parameter.